### PR TITLE
feat(bigtable): Restored microseconds as timestamp range filter ranges

### LINKED ISSUE #5931: bigtable: Allow microseconds for TimestampRangeFilter (again)
**Is your feature request related to a problem? Please describe.**
Could you please remove code rejecting microseconds when defining `TimestampRangeFilter`s?

**Describe the solution you'd like**
1. Remove if-block [here](https://github.com/googleapis/google-cloud-go/blob/main/bigtable/bttest/inmem.go#L746-L753)
2. Update tests [here](https://github.com/googleapis/google-cloud-go/blob/f37584ff8947b18332b214a958ac57010cb6df5b/bigtable/bttest/inmem_test.go#L1721-L1722) (remove these 2 lines)

**Describe alternatives you've considered**
Detect which API version is used (v1 vs. v2) and deny or allow microseconds for `TimestampRangeFilter` accordingly.

**Additional context**
It seems v2 APIs/clients use microseconds now when defining `TimestampRangeFilter`s. This breaks most tests we have after upgrading gcp packages to get our dev environments working on Mac M1.

---- 
The v2 proto defines TimestampFilter using micros:
```proto
# google.bigtable.v2.data:
// Specified a contiguous range of microsecond timestamps.
message TimestampRange {
  // Inclusive lower bound. If left empty, interpreted as 0.
  int64 start_timestamp_micros = 1;

  // Exclusive upper bound. If left empty, interpreted as infinity.
  int64 end_timestamp_micros = 2;
}
```

and here is `Filters.java` coming from `com.google.cloud.bigtable.data.v2.models` package:

```java
        public static final class TimestampRangeFilter
      extends AbstractTimestampRange<TimestampRangeFilter> implements Filter {
    ...
    public RowFilter toProto() {
      com.google.bigtable.v2.TimestampRange.Builder builder =
          com.google.bigtable.v2.TimestampRange.newBuilder();

      switch (getStartBound()) {
        case CLOSED:
          builder.setStartTimestampMicros(getStart());
          break;
        case OPEN:
          builder.setStartTimestampMicros(getStart() + 1);
          break;
        case UNBOUNDED:
          break;
        default:
          throw new IllegalStateException("Unknown start bound: " + getStartBound());
      }
```

### PR DESCRIPTION
**Pull request by mmetto**
fixes #5931

### CODE REVIEW

**Comment by mmetto** [comment]
Hello @telpirion @igorbernstein2, kindly reminding about this small PR. Thanks in advance for taking your time to look at it.

**Comment by mmetto** [comment]
Ping! Can we get traction on this issue? Thanks in advance!

**Comment by mmetto** [comment]
ping, can I get traction on this issue please? @igorbernstein2 @telpirion

**Comment by mmetto** [comment]
ping, can we get traction on this issue please? @igorbernstein2 @telpirion

**Comment by mmetto** [comment]
@telpirion @igorbernstein2 can I get your eyes on this change? This is blocking our testing and requires manual changes to get the tests working properly with newest bigtable client/models locally, while also locking us to use old version of dependencies for our service.
If you're not happy with the changes introduced, please feel free to reach out to me and let's get things worked out.

**Comment by mmetto** [comment]
Ping, is there any, even remote chance we'll get traction on this PR?

**Comment by telpirion** [comment]
@mmetto Yep, let's see if we can get this PR taken care of. Thank you for the contribution!

File: bigtable/bttest/inmem.go [hunk 1]
```diff
@@ -972,10 +968,6 @@ func (s *server) CheckAndMutateRow(ctx context.Context, req *btpb.CheckAndMutate
 	return res, nil
 }
 
-func (s *server) PingAndWarm(ctx context.Context, req *btpb.PingAndWarmRequest) (*btpb.PingAndWarmResponse, error) {
```
Comment (telpirion) [inline bigtable/bttest/inmem.go:975]: Why remove `PingAndWarm()`?

Comment (mmetto) [reply bigtable/bttest/inmem.go:975]: Erm, not sure how it ended up being removed, it's been a while but I'm sure I only touched lines 748-751 in this file.

Let me also rebase this branch.

Comment (mmetto) [reply bigtable/bttest/inmem.go:981]: well, git history says I removed it...

Comment (mmetto) [reply bigtable/bttest/inmem.go]: @telpirion reverted that change in https://github.com/googleapis/google-cloud-go/pull/5973/commits/60042f3c1898726f0a75af68c895f017d1dab17b, please let me know if there are any other comments to address, thanks in advance!

**Comment by igorbernstein2** [comment]
Sorry for missing this, but bigtable service will reject microseconds that are not a multiple of 1000. So removing these checks from the emulator is counterproductive

**Comment by telpirion** [comment]
Given that this PR would give the emulator a behavior different than the service, I'm going to close.

**Pull request closed without merging**